### PR TITLE
sensors/vehicle_gps_position: untangle and remove unnecessary state

### PR DESCRIPTION
**Pull request by dagar**
@JacobCrabill @bresch this is the refactor/cleanup of GPS blending originally from https://github.com/PX4/Firmware/pull/14447.

NOT TESTED.

This would be a good excuse to add proper testing in SITL for GPS blending and perhaps even restructure the blending logic to make it unit testable.

### CODE REVIEW

**Comment by dagar** [comment]
Relatedly, now that GPS blending has moved to the sensors hub we can think about expanding it slightly for Multi-EKF (https://github.com/PX4/Firmware/pull/14650).

 - SENS_GPS_MODE
    - publish all GPS, publish a specific instance, publish blended, publish highest priority, etc

 We also need to add per instance metadata at the sensors level instead of single set of position offsets in EKF2.
   - EKF2_GPS_POS_{X,Y,Z} -> SENS_GPS_n_POS_{X,Y,Z} (FYI @CarlOlsson)
   - to keep things simple we can pass the position in `vehicle_gps_position` since the message is low rate (5-20 Hz)
   - GPS will need device ids

File: src/modules/sensors/vehicle_gps_position/VehicleGPSPosition.cpp [hunk 1]
```diff
@@ -490,86 +508,83 @@ void VehicleGPSPosition::update_gps_blend_states()
...
+	float alpha[GPS_MAX_RECEIVERS] {};
 	float omega_lpf = 1.0f / fmaxf(_param_sens_gps_tau.get(), 1.0f);
 
 	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
 		if (_gps_state[i].timestamp - _time_prev_us[i] > 0) {
 			// calculate the filter coefficient that achieves the time constant specified by the user adjustable parameter
-			float min_alpha = constrain(omega_lpf * 1e-6f * (float)(_gps_state[i].timestamp - _time_prev_us[i]),
-						    0.0f, 1.0f);
-
-			// scale the filter coefficient so that time constant is inversely proprtional to weighting
```
Comment (bresch) [inline src/modules/sensors/vehicle_gps_position/VehicleGPSPosition.cpp:566]: isn't that useful?

Comment (dagar) [reply src/modules/sensors/vehicle_gps_position/VehicleGPSPosition.cpp:566]: It looks like it was removed back in ekf2, but somehow got lost in the drawn out PR sitting around for 6 months.

https://github.com/PX4/PX4-Autopilot/commit/8de1f5229bd70cccaa8eb2605fbfb6a17b304176#diff-72bb106f8fdfef0de0e8174f5b63ad2987bd47a8f07ff445bde40a812c87651e

**Comment by dagar** [comment]
A lot of the state has been removed, but this is still hard to follow. Let's layout a simple test plan here.

**Comment by JacobCrabill** [comment]
Just took a brief look through, and seems ok - I'll give myself a TODO to try it out with a couple GPSs on a real board, and capture logs.

**Comment by JacobCrabill** [comment]
Blending looks good (using dual Here3 GPS units over UAVCAN) (rebased just before testing to ensure the UAVCAN bridge would work):
image
image

**Comment by JacobCrabill** [comment]
@dagar to be clear, all I did for testing was ensure that two `sensor_gps` publications were combined in a convex combination into the `vehicle_gps_position` topic; not exactly thorough, but more of a sanity check.

**Comment by dagar** [comment]
Thanks @JacobCrabill
